packages/grafana-runtime/src/utils/analytics.ts

```diff
@@ -43,13 +43,20 @@ export const reportPageview = () => {
  * @public
  */
 export const reportInteraction = (interactionName: string, properties?: Record<string, any>) => {
-  getEchoSrv().addEvent<InteractionEchoEvent>({
-    type: EchoEventType.Interaction,
-    payload: {
+  if (process.env.NODE_ENV === 'production') {
```

Comment:
Only log when in prod, otherwise just console.log the event .

I found it useful for testing on localhost

Reply:
Would you mind putting this in a separate PR? 🤞🏻 

This PR already has a reasonably high impact, and I think this change is big and separate enough to warrant it's own PR and discussion.

Reply:
My gut feeling is this is not how we should implement logging interactions. Instead I think echo service should have a "console log backend" added, which can be turned on or off depending on configuration/feature flag/idk

Reply:
Absolutely, that makes sense, especially as I wanted to console.log on `test` too but then that ended up with a bunch of unit tests complaining about console logs 😉 !

I really think we should not emit events to Rudderstack on dev or test, but happy to look into the "console log backend" implementation for that 👍 

I'll remove from this PR as it is absolutely not required for the event 🙂

Reply:
FYI events will only go to rudderstack in _any_ environment (prod, dev, cloud, on-prem) if you enable it https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#rudderstack_write_key

Reply:
My plan was to simplify the code here with `if prod -> trigger event else -> console.log` and to mock `reportInteraction` in the complaining tests (so there would be no console output). But I also understand your point, since a test should *not* output to the console, why include `test` in here at all, I might go back to `if dev -> console.log, else -> trigger event`.